### ISSUE TITLE
Correct code location for table driven tests

### ISSUE THREAD

**Issue by hui-yang** [onsi/ginkgo#515]
For table driven tests, the code location is always showing https://github.com/onsi/ginkgo/blob/master/extensions/table/table_entry.go#L46 which is not correct. This should be a useful feature for some testers.

Is it possible to call PutItNode directly with correct call depth like this? https://github.com/onsi/ginkgo/blob/master/ginkgo_dsl.go#L375

**Comment by nodo** [comment]
Thanks @hui-yang,

Thanks very interesting feedback!

I have prepared a small reproducer:

```go
package bar_test

import (
        . "github.com/onsi/ginkgo"
        . "github.com/onsi/ginkgo/extensions/table"
        . "github.com/onsi/gomega"
)

var _ = Describe("Bar", func() {
        DescribeTable("table",
                func(i int) {
                        Expect(i).To(Equal(0))
                },
                Entry("yolo1", 0),
                Entry("yolo2", 0),
                Entry("yolo3", 1),
                Entry("yolo4", 0),
                Entry("yolo5", 0),
        )
})
```

Currently outputs:
```
Running Suite: Bar Suite
========================
Random Seed: 1533142767
Will run 5 of 5 specs

••
------------------------------
• Failure [0.001 seconds]
Bar
/Users/andreanodari/go/src/github.com/nodo/bar/bar_test.go:9
  table
  /Users/andreanodari/go/src/github.com/onsi/ginkgo/extensions/table/table.go:92
    yolo3 [It]
    /Users/andreanodari/go/src/github.com/onsi/ginkgo/extensions/table/table_entry.go:46

    Expected
        <int>: 1
    to equal
        <int>: 0

    /Users/andreanodari/go/src/github.com/nodo/bar/bar_test.go:12
------------------------------
```

- It mentions "yolo3" as failing entry
- It mentions line 12, which points to the failing "Expect"

However:
- It does not mention the line of the failing `Entry`. Which could potentially be useful information.

Would you be interested in working on a PR?

**Comment by ansd** [comment]
Fixed in #666
